bigquery/snippets/snippet.go

```diff
@@ -407,6 +453,194 @@ func createTableExplicitSchema(client *bigquery.Client, datasetID, tableID strin
 	return nil
 }
 
+func createTableComplexSchema(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_nested_repeated_schema]
+	sampleSchema := bigquery.Schema{
+		{Name: "id", Type: bigquery.StringFieldType},
+		{Name: "first_name", Type: bigquery.StringFieldType},
+		{Name: "last_name", Type: bigquery.StringFieldType},
+		{Name: "dob", Type: bigquery.DateFieldType},
+		{Name: "addresses",
+			Type:     bigquery.RecordFieldType,
+			Repeated: true,
+			Schema: bigquery.Schema{
+				{Name: "status", Type: bigquery.StringFieldType},
+				{Name: "address", Type: bigquery.StringFieldType},
+				{Name: "city", Type: bigquery.StringFieldType},
+				{Name: "state", Type: bigquery.StringFieldType},
+				{Name: "zip", Type: bigquery.StringFieldType},
+				{Name: "numberOfYears", Type: bigquery.StringFieldType},
+			}},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	if err := tableRef.Create(ctx, metaData); err != nil {
+		return err
+	}
+	fmt.Printf("created table %s\n", tableRef.FullyQualifiedName())
+	// [END bigquery_nested_repeated_schema]
+	return nil
+}
+
+func createTableWithCMEK(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_create_table_cmek]
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	meta := &bigquery.TableMetadata{
+		EncryptionConfig: &bigquery.EncryptionConfig{
+			// TODO: Replace this key with a key you have created in Cloud KMS.
+			KMSKeyName: "projects/cloud-samples-tests/locations/us-central1/keyRings/test/cryptoKeys/test",
+		},
+	}
+	if err := tableRef.Create(ctx, meta); err != nil {
+		return err
+	}
+	// [END bigquery_create_table_cmek]
+	return nil
+}
+
+func relaxTableAPI(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_relax_column]
+	sampleSchema := bigquery.Schema{
+		{Name: "full_name", Type: bigquery.StringFieldType, Required: true},
+		{Name: "age", Type: bigquery.IntegerFieldType, Required: true},
+	}
+	original := &bigquery.TableMetadata{
+		Schema: sampleSchema,
+	}
+	tableRef := client.Dataset(datasetID).Table(tableID)
+	if err := tableRef.Create(ctx, original); err != nil {
```

Comment:
A table Get might be more natural for this sample (with the create outside the sample).

Reply:
It would simplify the other point about schema evolution, will do that instead.

Reply:
s/that/this/